scripts/pylib/twister/twisterlib/size_calc.py

```diff
@@ -160,18 +147,71 @@ def get_rom_size(self):
         """
         return self.rom_size
 
+    def get_percentage_usage_ram(self):
+        """Get the percentage usage of the data that this application uses on device
+
+        @return percentage of RAM.
+        """
+        return self.percentage_usage_ram
+
+    def get_percentage_usage_rom(self):
+        """Get the percentage usage of the data that this application uses on device's flash
+
+        @return percentage of ROM.
+        """
+        return self.percentage_usage_rom
+
     def unrecognized_sections(self):
         """Get a list of sections inside the binary that weren't recognized
 
         @return list of unrecognized section names
         """
         slist = []
-        for v in self.sections:
-            if not v["recognized"]:
-                slist.append(v["name"])
+        if self.filename.endswith(".elf"):
+            for v in self.sections:
+                if not v["recognized"]:
+                    slist.append(v["name"])
         return slist
 
     def _calculate_sizes(self):
+        if self.filename.endswith("build.log"):
+            self._get_footprint_from_buildlog(pattern_information_line=self.PATTERN_SEARCHED_LINE)
```

Comment:
`pattern_information_line` is passed multiple times from one method to another. Please consider if this will be simpler if information about this pattern could be stored just as class attribute and initialize in `__init__`:
```python
def __init__(self, filename, extra_sections)
    ...
    self.pattern_information_line = self.PATTERN_SEARCHED_LINE
    ...
```

Reply:
Done